### PR TITLE
Update the site theme to work for child themes

### PR DESCRIPTION
**Pull request by jameswburke**
## Summary
As titled.

## Notes for reviewers
Still needs unit tests.

## Changelog entries
* **Added**: Support for child themes using the parent's site theme values.

## Ticket(s)
https://alleyinteractive.atlassian.net/browse/IRV-980

### CODE REVIEW

**Review by montchr — CHANGES_REQUESTED** [review]
Just one blocker – see my comment about the `wp_irving_site_theme_json_directory_path` filters. Otherwise, looks good.

File: inc/components/namespace.php [hunk 1]
```diff
@@ -70,8 +69,9 @@ function auto_register_components() {
 	$directories = apply_filters(
 		'wp_irving_component_registry_directories',
 		[
-			'wp_irving' => WP_IRVING_PATH . '/inc/components/components', // Load components from WP Irving.
-			'theme'     => get_stylesheet_directory() . '/components/', // Load components from the activated theme.
+			'wp_irving'     => WP_IRVING_PATH . '/inc/components/components', // Load components from WP Irving.
+			'current-theme' => get_stylesheet_directory() . '/components/', // Load components from the activated theme.
+			'parent-theme'  => get_template_directory() . '/components/', // Load components from any parent theme.
 		]
 	);
```
[the remark this answers is not in the document]
Comment (jameswburke) [reply inc/components/namespace.php:77]: Thanks, addressed in https://github.com/alleyinteractive/wp-irving/pull/317/files#diff-de08f947e3462eccc1d88d520847dc42c66d3d1809fe8536b0c10443d8c0bcb6R92-R95.

File: inc/templates/site-theme.php [hunk 2]
```diff
@@ -101,20 +101,59 @@ function get_site_theme( string $selector = '', $default = null ) {
...
  */
 function get_site_theme_from_json_files() {
 
-	$theme = [];
-
-	$path = apply_filters( 'wp_irving_site_theme_json_directory_path', get_stylesheet_directory() . '/styles/' );
+	// Default paths for parent and child themes, where the child theme takes precedence.
+	$default_paths = [
+		apply_filters( 'wp_irving_site_theme_json_directory_path', get_template_directory() . '/styles/' ), // Support parent themes.
+		apply_filters( 'wp_irving_site_theme_json_directory_path', get_stylesheet_directory() . '/styles/' ), // Support child themes.
```
[the remark this answers is not in the document]
Comment (jameswburke) [reply inc/templates/site-theme.php]: Reworked the way this is handled. I was trying to avoid breaking changes, but it was confusing.

**Review by montchr — APPROVED** [review]
🌀

**Pull request merged**